micro_dl/inference/image_inference.py

```diff
@@ -669,94 +672,100 @@ def predict_2d(self, chan_slice_meta):
-                    cur_input,
-                    self.crop_shape,
-                    self.image_format,
-                )
-                cur_target = image_utils.center_crop_to_shape(
-                    cur_target,
-                    self.crop_shape,
-                    self.image_format,
-                )
-            if self.tile_option == 'tile_xy':
-                print('tiling input...')
-                step_size = (np.array(self.tile_params['tile_shape']) -
-                             np.array(self.num_overlap))
-
-                # TODO tile_image works for 2D/3D imgs, modify for multichannel
-                if self.data_format == 'channels_first':
-                    cur_input_1chan = cur_input[0, 0, ...]
+        with tqdm(total=len(chan_slice_meta['slice_idx'].unique()), desc='z-stack prediction', leave=False) as pbar:
+            for z_idx in chan_slice_meta['slice_idx'].unique():
+                chan_meta = chan_slice_meta[chan_slice_meta['slice_idx'] == z_idx]
+                cur_input, cur_target = \
+                    self.dataset_inst.__getitem__(chan_meta.index[0])
+                if self.crop_shape is not None:
+                    cur_input = image_utils.center_crop_to_shape(
+                        cur_input,
+                        self.crop_shape,
+                        self.image_format,
+                    )
+                    cur_target = image_utils.center_crop_to_shape(
+                        cur_target,
+                        self.crop_shape,
+                        self.image_format,
+                    )
+                if self.tile_option == 'tile_xy':
+                    print('tiling input...')
+                    step_size = (np.array(self.tile_params['tile_shape']) -
+                                 np.array(self.num_overlap))
+
+                    # TODO tile_image works for 2D/3D imgs, modify for multichannel
+                    if self.data_format == 'channels_first':
+                        cur_input_1chan = cur_input[0, 0, ...]
+                    else:
+                        cur_input_1chan = cur_input[0, ..., 0]
+                    _, crop_indices = tile_utils.tile_image(
+                        input_image=np.squeeze(cur_target),
+                        tile_size=self.tile_params['tile_shape'],
+                        step_size=step_size,
+                        return_index=True
+                    )
+                    print('crop_indices:', crop_indices)
+                    pred_block_list = self._predict_sub_block_xy(
+                        cur_input,
+                        crop_indices,
+                    )
+                    pred_image = self.stitch_inst.stitch_predictions(
+                        cur_target[0].shape,
+                        pred_block_list,
+                        crop_indices,
+                    )
+                    # add batch dimension
+                    pred_image = pred_image[np.newaxis, ...]
```

Comment:
You probably need to add an extra dimension even if you don't use this specific tile option, no?